src/NativeCameraModule.ts

```diff
@@ -0,0 +1,41 @@
+import { NativeModules, Platform } from 'react-native';
+import { CameraRuntimeError } from './CameraError';
+
+const supportedPlatforms = ['ios', 'android', 'macos'];
+
+// NativeModules automatically resolves 'CameraView' to 'CameraViewModule'
+// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+export const CameraModule = NativeModules.CameraView;
+if (CameraModule == null) {
+  if (!supportedPlatforms.includes(Platform.OS)) {
+    throw new CameraRuntimeError(
+      'system/camera-module-not-found',
+      `Failed to initialize VisionCamera: VisionCamera currently does not work on ${Platform.OS}.`,
+    );
+  }
+
+  let message = 'Failed to initialize VisionCamera: The native Camera Module (`NativeModules.CameraView`) could not be found.';
+  message += '\n* Make sure react-native-vision-camera is correctly autolinked (run `npx react-native config` to verify)';
+  if (Platform.OS === 'ios' || Platform.OS === 'macos') message += '\n* Make sure you ran `pod install` in the ios/ directory.';
+
+  if (Platform.OS === 'android') message += '\n* Make sure gradle is synced.';
+
+  // check if Expo
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  const ExpoConstants = NativeModules.NativeUnimoduleProxy?.modulesConstants?.ExponentConstants;
```

Comment:
In the future `NativeModules.NativeUnimoduleProxy` will become deprecated and removed. Now all Expo modules are installed through JSI and accessible from `global.expo.modules` object so something like this might be better 👇 
```suggestion
  const ExpoConstants = global.expo?.modules?.ExponentConstants;
```

Reply:
Thanks Tomasz, changed this on v3! 👍 

Out of curiosity, since which Expo version is this supported? Do I need to think about backwards compatibility with the `NativeUnimodulesProxy` if VisionCamera only runs on RN 71 and above anyways, or no?

Reply:
If that's >0.71 then I guess you don't need to worry about that. `global.expo.modules` was added in SDK 47 in which the recommended RN version is 0.70 😉